test/yield_component_token.ts

```diff
@@ -204,18 +204,108 @@ describe("YieldComponentToken", () => {
+      // expect(await erc20Token.balanceOf(sender)).to.eq(0);
+      // await yieldComponentToken.connect(senderSigner).transfer(owner, "1000000");
+      // // no payout happens without a price increase
+      // expect(await erc20Token.balanceOf(sender)).to.eq(0);
+      // await priceOracle.setPrice("1200000000");
+      // await yieldComponentToken.connect(senderSigner).transfer(owner, "1000000");
+      // // price has increased, and so a transfer should trigger a real payout
+      // expect(await erc20Token.balanceOf(sender)).to.eq(0);
+    });
+    it("should payout yield to both sender and receiver when both have balances", async () => {});
   });
-  describe("transfer", async () => {
+  describe("transferFrom", async () => {
+    it("should not allow a transfer when an allowance has not been set", async () => {
+      const [ownerSigner, senderSigner] = await ethers.getSigners();
+      const [owner, sender] = await Promise.all([ownerSigner.getAddress(), senderSigner.getAddress()]);
+      const amountToSend = "12340000";
+      const { yieldComponentToken } = await deployAll("X Token", "XXX", deployedAddresses, splitVault);
+      await yieldComponentToken.mint(owner, "10000000000000");
+      await expect(
+        yieldComponentToken.connect(senderSigner).transferFrom(owner, sender, amountToSend),
+      ).to.be.revertedWith("ERC20: transfer amount exceeds allowance");
+    });
+    it("should not allow a transfer when an allowance is too small", async () => {
+      const [ownerSigner, senderSigner] = await ethers.getSigners();
+      const [owner, sender] = await Promise.all([ownerSigner.getAddress(), senderSigner.getAddress()]);
+      const amountToSend = "12340000";
+      const { yieldComponentToken } = await deployAll("X Token", "XXX", deployedAddresses, splitVault);
+      await yieldComponentToken.mint(owner, "10000000000000");
+      await yieldComponentToken.approve(sender, "1234");
+      await expect(
+        yieldComponentToken.connect(senderSigner).transferFrom(owner, sender, amountToSend),
+      ).to.be.revertedWith("ERC20: transfer amount exceeds allowance");
+    });
+    it("should allow a transfer when an allowance is set and is great enough", async () => {
+      const [ownerSigner, senderSigner] = await ethers.getSigners();
+      const [owner, sender] = await Promise.all([ownerSigner.getAddress(), senderSigner.getAddress()]);
+      const amountToSend = "12340000";
+      const { yieldComponentToken } = await deployAll("X Token", "XXX", deployedAddresses, splitVault);
+      await yieldComponentToken.mint(owner, "10000000000000");
+      await yieldComponentToken.approve(sender, amountToSend);
+      expect(await yieldComponentToken.balanceOf(sender)).to.eq(0);
+      await yieldComponentToken.connect(senderSigner).transferFrom(owner, sender, amountToSend);
+      expect(await yieldComponentToken.balanceOf(sender)).to.eq(amountToSend);
```

Comment:
Nit: It's awkward that this test involves sending tokens to the `sender` address. Maybe introduce a third EOA, maybe named `receiver`?